commands/commands.spec.ts

```diff
@@ -14,17 +14,110 @@ describe('Commands', () => {
     expect(cmds).toBeTruthy()
   })
 
+  it('responds to nonsense commands', async () => {
+    let mockNode: any = jest.fn()
+    expect(mod.Commands).toBeDefined()
+    let cmds = new mod.Commands(mockNode)
+
+    expect(await cmds.execute('not-a-real-command')).toMatch(/unknown command/i)
+  })
+
   it('ping', async () => {
     let mockNode: any = jest.fn()
     mockNode.bootstrapServers = []
     mockNode.ping = jest.fn()
     let mockPeerId = '16Uiu2HAkyXRaL7fKu4qcjaKuo4WXizrpK63Ltd6kG2tH6oSV58AW'
 
     let cmds = new mod.Commands(mockNode)
-    await cmds.execute(`ping  ${mockPeerId}`)
+    expect(await cmds.execute(`ping  ${mockPeerId}`)).toMatch(/pong/i)
     expect(mockNode.ping).toHaveBeenCalled()
+  })
+
+
+  it('commands can save state', async () => {
+    let mockNode: any = jest.fn()
+    let cmds = new mod.Commands(mockNode)
+
+    let ir = await cmds.execute('settings includeRecipient')
+    expect(ir).toMatch(/false/)
+    await cmds.execute('includeRecipient true')
+    ir = await cmds.execute('settings includeRecipient')
+    expect(ir).toMatch(/true/)
+    await cmds.execute('includeRecipient false')
+    ir = await cmds.execute('settings includeRecipient')
+    expect(ir).toMatch(/false/)
+  })
+
+  it('send message', async () => {
+    let mockNode: any = jest.fn()
+    mockNode.sendMessage = jest.fn()
+    let cmds = new mod.Commands(mockNode)
+    await cmds.execute('send 16Uiu2HAmAJStiomwq27Kkvtat8KiEHLBSnAkkKCqZmLYKVLtkiB7 Hello, world')
+    expect(mockNode.sendMessage).toHaveBeenCalled()
+  })
+
+  it('alias addresses', async () => {
+    let mockNode: any = jest.fn()
+    mockNode.sendMessage = jest.fn()
+    let cmds = new mod.Commands(mockNode)
+
+    let aliases = await cmds.execute('settings alias')
+    expect(aliases).toEqual('')
+
+    await cmds.execute('alias 16Uiu2HAmQDFS8a4Bj5PGaTqQLME5SZTRNikz9nUPT3G4T6YL9o7V test')
+
+    aliases = await cmds.execute('settings alias')
+    expect(aliases).toMatch(/test/)
+    await cmds.execute('send test Hello, world')
+    expect(mockNode.sendMessage).toHaveBeenCalled()
```

Comment:
Might be a silly question here, but I guess that even if `sendMessage` fails (i.e. sending it to an invalid address), wouldn't it mean that it was called anyway? Wouldn't be better to make sure that it didn't throw an error?

Reply:
This just means that it has called sendMessage, not that it's passing the correct arguments, or anything else really. It's a mock function, so it definitely won't throw.